hnn_core/network.py

```diff
@@ -254,6 +256,21 @@ def _add_external_feeds(self):
         # Create the feed dynamics (event_times)
         self._instantiate_feeds(n_trials=self.params['N_trials'])
 
+        # Add tonic inputs if present
+        for cell_type in ['L2Basket', 'L5Basket', 'L2Pyr', 'L5Pyr']:
+            is_tonic_present = [f'Itonic_{p}_{cell_type}' in self.params
+                                for p in ['A', 't0', 'T']]
+            if any(is_tonic_present) and not all(is_tonic_present):
+                raise ValueError(f'Tonic input must have the amplitude, '
+                                 f'start time and end time specified. One '
+                                 f'or more parameter may be missing for '
+                                 f'cell type {cell_type}')
+            self.feed_times['tonic'][cell_type] = {
+                'amplitude': self.params[f'Itonic_A_{cell_type}'],
+                't0': self.params[f'Itonic_t0_{cell_type}'],
+                'T': self.params[f'Itonic_T_{cell_type}']
```

Comment:
So this is a very different use of `self.feed_times` compared with what's done in `_instantiate_feeds`, but makes sense since tonic inputs have NEURON'ism to them, so the instantiation _has_ to wait until `NetworkBuilder`.

But what about `trial_idx`? Shouldn't this definition have `self.feed_times['tonic'][trial_idx][cell_type]`, just like the "regular" external drives?

Reply:
yeah tonic drives are not allowed to vary between trials. I could have replicated the dictionary over `n_trials` but that just seemed like adding unnecessary complexity.

Reply:
At the risk of finding a can of worms: _why_ aren't they allowed to vary? We're not really dealing with a "tonic" input, but a Poisson process with random dynamics. As a naive user I'd be very surprised if two independent trials yielded the same process, and the same dynamics in the gamma examples, for example.

Reply:
humm ... I'm not sure why you say it's a Poisson process? We are using a [current clamp](https://www.neuron.yale.edu/neuron/static/py_doc/modelspec/programmatic/mechanisms/mech.html#IClamp) to provide a pulsed input

Reply:
Ach, my bad! Totally agree makes no sense to allow to vary across trials then.